### ISSUE TITLE
Crash reports for PW Activities

### ISSUE THREAD

**Issue by khltrifork** [Pushwoosh/pushwoosh-android-sdk#124]
I have crash tracking in production system, and every now and then we get one of these types of crashes:

```
Exception: java.lang.RuntimeException

Unable to start activity ComponentInfo{xx.yy.zz/com.pushwoosh.inapp.view.RichMediaWebActivity}: java.lang.NullPointerException
android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2118)
android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2145)
android.app.ActivityThread.access (ActivityThread.java:130)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1211)
android.os.Handler.dispatchMessage (Handler.java:99)
android.os.Looper.loop (Looper.java:137)
android.app.ActivityThread.main (ActivityThread.java:4847)
java.lang.reflect.Method.invokeNative (Native Method)
java.lang.reflect.Method.invoke (Method.java:535)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:786)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:553)
dalvik.system.NativeStart.main (Native Method)

Caused By: java.lang.NullPointerException
android.view.ViewGroup.addViewInner (ViewGroup.java:3389)
android.view.ViewGroup.addView (ViewGroup.java:3251)
android.view.ViewGroup.addView (ViewGroup.java:3227)
com.android.internal.policy.impl.PhoneWindow.setContentView (PhoneWindow.java:289)
com.android.internal.policy.impl.PhoneWindow.setContentView (PhoneWindow.java:279)
android.app.Activity.setContentView (Activity.java:1999)
com.pushwoosh.inapp.view.RichMediaWebActivity.a (Unknown Source)
com.pushwoosh.inapp.view.WebActivity.a (Unknown Source)
com.pushwoosh.inapp.view.WebActivity.onCreate (Unknown Source)
com.pushwoosh.inapp.view.RichMediaWebActivity.onCreate (Unknown Source)
android.app.Activity.performCreate (Activity.java:5187)
android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1104)
android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2064)

```

And this:

```
Exception: java.lang.RuntimeException

Unable to start activity ComponentInfo{xx.yy.zz/com.pushwoosh.inapp.view.RemoteUrlActivity}: java.lang.NullPointerException
android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2118)
android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2145)
android.app.ActivityThread.access (ActivityThread.java:130)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1211)
android.os.Handler.dispatchMessage (Handler.java:99)
android.os.Looper.loop (Looper.java:137)
android.app.ActivityThread.main (ActivityThread.java:4847)
java.lang.reflect.Method.invokeNative (Native Method)
java.lang.reflect.Method.invoke (Method.java:535)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:786)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:553)
dalvik.system.NativeStart.main (Native Method)

Caused By: java.lang.NullPointerException
com.pushwoosh.inapp.view.WebActivity.a (Unknown Source)
com.pushwoosh.inapp.view.WebActivity.onCreate (Unknown Source)
android.app.Activity.performCreate (Activity.java:5187)
android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1104)
android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2064)
```

From what i can see this is cause when these activities are opened without the correct parameters, and this causes them to crash completely.

Would it be possible to simply check if the activity intent extras are valid before continuing, and if not valid, simply close the activity?

The device that is causing this is named "unknown Full Android on Emulator". I'm assuming it's some bot-device that tries to run all activities it can find in the manifest.

**Comment by wfhm** [comment]
Hi @khltrifork,

Could you please specify the SDK version affected?

**Comment by khltrifork** [comment]
We're currently using 5.9.3 in production (where this crash occurred).
That was the latest version available when we pushed a new version of the app into production.

Internally we have updated to 5.13.2, but this is not in production yet.

**Comment by wfhm** [comment]
@khltrifork This issue is already fixed in 5.12.1, so please try updating and let us know if it persists.
